app/src/tests/components/contents/downloads/FileRow.test.tsx

```diff
@@ -1,24 +1,29 @@
 import { render, screen } from "@testing-library/react";
 import { mockPacket } from "../../../mocks";
 import { FileRow } from "../../../../app/components/contents/downloads/FileRow";
-import { createMemoryRouter, Outlet, RouterProvider } from "react-router-dom";
+import { MemoryRouter, Outlet, Route, Routes } from "react-router-dom";
+import { FileMetadata } from "../../../../types";
+import { SWRConfig } from "swr";
 
 const expectIconToBeRendered = (container: HTMLElement, iconName: string) => {
   const icon = container.querySelector(".lucide") as HTMLImageElement;
   expect(icon.classList).toContain(iconName);
 };
 
-const renderComponent = (path: string, sharedResource?: boolean) => {
-  const routes = [
-    {
-      path: "/",
-      element: <Outlet context={{ packet: mockPacket }} />,
-      children: [{ path: "/", element: <FileRow path={path} sharedResource={sharedResource} /> }]
-    }
-  ];
-  const router = createMemoryRouter(routes, { initialEntries: ["/"] });
-
-  return render(<RouterProvider router={router} />);
+const renderComponent = (filePath: string, sharedResource?: boolean) => {
+  const file = mockPacket.files.find((file) => file.path === filePath) as FileMetadata;
+
+  return render(
+    <SWRConfig value={{ dedupingInterval: 0 }}>
+      <MemoryRouter initialEntries={["/"]}>
+        <Routes>
+          <Route element={<Outlet context={{ packet: mockPacket }} />}>
+            <Route path="/" element={<FileRow file={file} sharedResource={sharedResource} />} />
+          </Route>
+        </Routes>
+      </MemoryRouter>
+    </SWRConfig>
+  );
```

Comment:
There are a couple of changes of this form. These give tests a more consistent renderComponent set-up when they need to be providing Outlet context.